### PR TITLE
UIP-47: Updating clustergrammer widget to v2 and un-skip clustergrammer test

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

This PR updates the clustergrammer widget to the new version.

Next PR will be another small one with an update to the Comm class that comes up in the pytest output.
I'll check on the various python updates needed and if they look straightforward, make a PR to update them, too.
I'll update the release notes and version once all this kerfuffle is done.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/UIP-47
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (Python) I have run Ruff `format` and `check` on changed Python code manually or with a git precommit hook

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: src/biokbase/narrative/tests/test_viewers.py [hunk 1]
```diff
@@ -20,13 +21,14 @@ def test_bad_view_as_clustergrammer_params_generic_ref(arg_type: str) -> None:
 
 @pytest.mark.vcr()
 def test_bad_view_as_clustergrammer_params_attr_set_ref() -> None:
+    """Test that the clustergrammer widget throws an error with invalid input."""
     with pytest.raises(ValueError, match="not a compatible data type"):
         viewers.view_as_clustergrammer(ATTRIBUTE_SET_REF)
 
 
 @pytest.mark.vcr()
-@pytest.mark.skip("Clustergrammer widget is currently broken")
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/test_viewers.py:28]: No skipping out on this test!

File: src/biokbase/narrative/viewers.py [hunk 2]
```diff
@@ -43,7 +40,7 @@ def view_as_clustergrammer(
 
     generic_df = get_df(ws_ref, col_categories, row_categories, True)
 
-    net = Network(clustergrammer_widget)
+    net = Network(CGM2)
```
Comment (ialarmedalien) [inline src/biokbase/narrative/viewers.py:43]: Updating the widget is kind of an anticlimax!

File: src/requirements.txt [hunk 3]
```diff
@@ -8,4 +8,5 @@ pymongo==4.7.0
 pyyaml==6.0.1
 requests==2.31.0
 scikit-learn==1.4.2
+statsmodels==0.14.2
```
Comment (ialarmedalien) [inline src/requirements.txt:11]: required for clustergrammer2 but not installed by default

File: src/biokbase/narrative/viewers.py (hunk 2 above)
Comment (briehl) [reply src/biokbase/narrative/viewers.py:43]: Better than rewriting everything!

**Review by briehl — APPROVED** [review]
🎉

**Pull request merged**